src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc

```diff
@@ -848,12 +848,14 @@ Real64 CoilCoolingDXCurveFitSpeed::calcEffectiveSHR(const DataLoopNode::NodeData
         }
     }
 
-    //  Use sucessive substitution to solve for To
+    //  Use successive substitution to solve for To
     aa = (Gamma * Toffa) - (0.25 / Twet) * pow_2(Gamma) * pow_2(Toffa);
     To1 = aa + Tcl;
     Error = 1.0;
     while (Error > 0.001) {
-        To2 = aa - Tcl * std::expm1(-To1 / Tcl);
+        //  Floating overflow errors occur when -To1/Tcl is a large positive number.
+        //  Cap upper limit at 700 to avoid the overflow errors.
+        To2 = aa - Tcl * std::expm1(min(700.0, -To1 / Tcl));
```

Comment:
@kevin-moos are you seeing an overflow or underflow? I have seen this limit (-700) used but for an [underflow condition here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc#L863-L866) and [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/DXCoils.cc#L12533-L12536). Tcl is the latent time constant which is a positive number. How did this calculation result in a large positive number? We do protect against large positive exponents (just search for 700) but not for this type of calculation. Please clarify.

Reply:
Is the fan operating as constant when this happens? The unit test shows cycling fan. The model is not supposed to be active during cycling fan.